docs/core/diagnostics/snippets/EventCounters/ConditionalEventCounterSource.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Diagnostics.Tracing;
+
+[EventSource(Name = "Samples.EventCounterDemos.Conditional")]
+public sealed class ConditionalEventCounterSource : EventSource
+{
+    public static readonly ConditionalEventCounterSource Log = new ConditionalEventCounterSource();
+
+    private EventCounter _requestCounter;
+
+    private ConditionalEventCounterSource() { }
+
+    protected override void OnEventCommand(EventCommandEventArgs args)
+    {
+        if (args.Command == EventCommand.Enable)
```

Comment:
For most consumer code doing conditional construction is probably unnecessary complexity. The runtime does it as a performance optimization to save a fraction of a millisecond on startup performance for scenarios that will never use the counters. I think it would be fine not to show this at all, but if we did I would add the context so that customers can decide if they care about performance optimizations that tiny.

Reply:
I'll add the additional context, thank you.